provisioning/transport/mqtt/src/mqtt.ts

```diff
@@ -52,11 +53,12 @@ export class Mqtt extends EventEmitter implements X509ProvisioningTransport, Sym
 
       /* Codes_SRS_NODE_PROVISIONING_MQTT_18_010: [ When waiting for responses, `registrationRequest` shall watch for messages with a topic named $dps/registrations/res/<status>/?$rid=<rid>.] */
       /* Codes_SRS_NODE_PROVISIONING_MQTT_18_024: [ When waiting for responses, `queryOperationStatus` shall watch for messages with a topic named $dps/registrations/res/<status>/?$rid=<rid>.] */
-      let match = topic.match(/^\$dps\/registrations\/res\/(.*)\/\?\$rid=(.*)$/);
+      let match = topic.match(/^\$dps\/registrations\/res\/(.*)\/\?(.*)$/);
+      let queryParameters = queryString.parse(match[2]);
 
-      if (!!match && match.length === 3) {
+      if (!!match && match.length === 3 && queryParameters['$rid']) {
```

Comment:
>queryParameters['$rid'] [](start = 43, length = 23)

accessing the $rid property like this causes linter errors

Reply:
Indeed it does.  Changing it to simply use .$rid